src/tests/App.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { render, fireEvent, screen, waitFor } from "@testing-library/react";
+import fetch from "jest-fetch-mock";
+import React from "react";
+import { Provider } from "react-redux";
+import { store } from "../redux/store";
+import App from "../App";
+import { RESULTS } from "./stubs";
+import { createStore } from "redux";
+import { rootReducer } from "../redux/reducers";
+import { Difficulty, fetchQuizQuestions } from "../API";
+
+describe("App", () => {
+
+
+  test("sucessful game flow", async () => {
+    render(
+      <Provider store={store}>
+        <App />
+      </Provider>
+    );
+    fetch.mockResponseOnce(JSON.stringify(RESULTS));
+
+    const startButton = screen.getByText("Start");
+
+    //Fetch test
+    expect(startButton).toBeVisible();
+    fireEvent.click(startButton);
+    expect(fetch).toHaveBeenCalledTimes(1);
+
+    //Loading state
+    expect(screen.getByText("Loading...")).toBeVisible();
+    expect(screen.getByText(/Score:/i)).toBeVisible();
+
+    //Loaded state
+    await waitFor(() => screen.getByTestId("questionCard"));
+    expect(screen.getByTestId("questionCard")).toBeVisible();
+  });
+
+
+
+  test("question card use", async () => {
+    
+    fetch.mockResponseOnce(JSON.stringify(RESULTS));
+    const result = await fetchQuizQuestions(2, Difficulty.Easy);
+    const newStore = createStore(
+      rootReducer,
+      {
+        game: {
```

Comment:
also, food for thought, but see here how your reducer has two levels
```
{
  game: {
   ...
  },
}
```

where first level only has one element, where you put everything in it.
That shows, that maybe that nesting is not needed at all. No reason root reducer couldn't be made up or a few tiny reducers (one for loading, one for questions, etc)